File: src/main/java/model/organization/event/UsesEventFactory.java
package model.organization.event;

import model.organization.relation.Uses;

/**
 * The {@linkplain UsesEventFactory} interface defines the API for constructing {@linkplain UsesEvent}s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface UsesEventFactory {

	/**
	 * Constructs an {@linkplain UsesEvent}.
	 *
	 * @param category
	 *            the {@linkplain EventCategory}.
	 * @param uses
	 *            the {@linkplain Uses}.
	 *
	 * @return a {@linkplain UsesEvent}.
	 */
	UsesEvent build(EventCategory category, Uses uses);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/relation/UsesEventFactory.kt
package runtimemodels.chazm.model.relation

import runtimemodels.chazm.api.relation.Uses
import runtimemodels.chazm.model.event.EventType

/**
 * The [UsesEventFactory] interface defines the API for constructing [UsesEvent]s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface UsesEventFactory {
    /**
     * Constructs an [UsesEvent].
     *
     * @param category the [EventType].
     * @param uses     the [Uses].
     * @return a [UsesEvent].
     */
    fun build(category: EventType, uses: Uses): UsesEvent
}
